pkg/services/app/add_test.go

```diff
@@ -0,0 +1,209 @@
+package app_test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/weaveworks/weave-gitops/pkg/flux/fluxfakes"
+	"github.com/weaveworks/weave-gitops/pkg/git/gitfakes"
+	"github.com/weaveworks/weave-gitops/pkg/gitproviders/gitprovidersfakes"
+	"github.com/weaveworks/weave-gitops/pkg/kube"
+	"github.com/weaveworks/weave-gitops/pkg/kube/kubefakes"
+	"github.com/weaveworks/weave-gitops/pkg/services/app"
+)
+
+var (
+	gitClient    *gitfakes.FakeGit
+	fluxClient   *fluxfakes.FakeFlux
+	kubeClient   *kubefakes.FakeKube
+	gitProviders *gitprovidersfakes.FakeGitProviderHandler
+
+	appSrv        app.AppService
+	defaultParams app.AddParams
+)
+
+var _ = BeforeEach(func() {
+	gitClient = &gitfakes.FakeGit{}
+	fluxClient = &fluxfakes.FakeFlux{}
+	kubeClient = &kubefakes.FakeKube{
```

Comment:
:partying_face: I dig this. Making the k8s client return what you want is a pain.